diff --git a/EnvironmentSimulator/code-examples/test-driver/test-driver.xosc b/EnvironmentSimulator/code-examples/test-driver/test-driver.xosc
index e4ee105..81e5c12 100644
--- a/EnvironmentSimulator/code-examples/test-driver/test-driver.xosc
+++ b/EnvironmentSimulator/code-examples/test-driver/test-driver.xosc
@@ -12,11 +12,11 @@
    </ParameterDeclarations>
    <CatalogLocations>
       <VehicleCatalog>
-         <Directory path="../resources/xosc/Catalogs/Vehicles"/>
+         <Directory path="./Catalogs/Vehicles"/>
       </VehicleCatalog>
    </CatalogLocations>
    <RoadNetwork>
-        <LogicFile filepath="../resources/xodr/curves_elevation.xodr"/>
+        <LogicFile filepath="../xodr/curves_elevation.xodr"/>
    </RoadNetwork>
    <Entities>
       <ScenarioObject name="Ego">
@@ -26,7 +26,7 @@
                 <Controller name="MyExternalControllerWithGhost">
                     <Properties>
                         <Property name="esminiController" value="ExternalController" />
-                        <Property name="useGhost" value="$GhostMode" />
+                        <Property name="useGhost" value="true" />
                         <Property name="headstartTime" value="2" />
                     </Properties>
                 </Controller>
